client/src/components/Chat/Messages/Content/ToolCallInfo.tsx

```diff
@@ -51,6 +54,23 @@ export default function ToolCallInfo({
         : localize('com_assistants_attempt_info');
   }
 
+  // Extract and parse ui_resources directly from the string
+  const uiResourcesMatch = output?.match(/<ui_resources>(.*?)<\/ui_resources>/s);
+  let uiResources: UIResource[] = [];
+
+  if (uiResourcesMatch && uiResourcesMatch[1]) {
+    try {
+      // Unescape the JSON string first
+      const unescapedJson = uiResourcesMatch[1].replace(/\\"/g, '"');
+      uiResources = JSON.parse(unescapedJson) as UIResource[];
+    } catch (error) {
+      console.error('Error parsing ui_resources:', error);
+    }
```

Comment:
Should we have something to fallback to? Will this break the UI?

Reply:
I'm going to test by sending badly formatted JSON, thanks!

Reply:
@mawburn actually this is what we get with a parsing error. I'm going to also wrap the `UI Resources` title conditionally upon parsing success, good catch man!

<img width="958" height="442" alt="image" src="https://github.com/user-attachments/assets/a5c77dc7-bff8-4b5d-a340-1579252c85f4" />